src/mapGenerator/ObjectGenerator.h

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2005 - 2015 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Return To The Roots.
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef ObjectGenerator_h__
+#define ObjectGenerator_h__
+
+#include "defines.h" // IWYU pragma: keep
+#include "mapGenerator/Map.h"
+#include "gameData/TerrainData.h"
+#include "Random.h"
+#include "stdint.h"
+#include <cstdlib>
+#include <utility>
+
+/**
+ * Utility class to place object, textures and animals on a map.
+ */
+class ObjectGenerator
+{
+    private:
+    
+    /**
+     * Generates a random number.
+     * @param max maximum value
+     * @return a new random number
+     */
+    static int Rand(const int max)
+    {
+        return rand() % max;
```

Comment:
The problem with the existing portable generator is the bad "randomness" of generated numbers for a large amount (which is required for map generation). Also, the way the map generation is implemented, it doesn't really matter if it is portable or not (apart from testability), because only the host is generating the map before the game starts.